sqlglot/dialects/dialect.py

```diff
@@ -1250,6 +1250,12 @@ def strposition_to_locate_sql(self: Generator, expression: exp.StrPosition) -> s
     )
 
 
+def strposition_to_instr_sql(self: Generator, expression: exp.StrPosition) -> str:
+    return self.func(
+        "INSTR", expression.args.get("substr"), expression.this, expression.args.get("position")
+    )
```

Comment:
Let's not implement this logic here. Instead, define `strposition_sql(self, expression: StrPosition) -> str:` within `SQLite.Generator`.

Additionally, isn't this supposed to be `INSTR(haystack, needle)`? You're generating three arguments here. Also, the relative order looks incorrect: `substr` comes first here, but you need it to be 2nd right?

Reply:
I've removed this altogether, since someone already implemented `dialect->str_position_sql`, which is also compatible with MySQL's `locate`.

And yes, the order was messed up.